lib/using_restricted_array.rb

```diff
@@ -4,60 +4,204 @@
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
+############################################################################
+
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n); requires n steps or n operations where n is the length of the array
+# Space complexity: O(n); requires n steps or n operations where n is the length of the array
+
 def length(array)
```

Comment:
👍 However your space complexity is O(1) since you are not building any other data structure.  Your method doesn't increase in memory usage if the input is bigger.